nautobot/ipam/models.py

```diff
@@ -709,7 +709,6 @@ def subnets(self, direct=False, include_self=False, for_update=False):
 
         return query.filter(
             ip_version=self.ip_version,
-            prefix_length__gte=self.prefix_length,
```

Comment:
With uniqueness constraints it would never be possible to have an overlapping network with the same prefix length and if the prefix length is shorter, either `network__gte` or `broadcast__lte` would not match.